### PR TITLE
fix(json-schema-2020-12): defaultModelExpandDepth not expanding schemas

### PR DESCRIPTION
**Pull request by glowcloud**
Refs #9508 
Supersedes #9510

### CODE REVIEW

File: test/e2e-cypress/static/pages/json-schema-2020-12-default-model-expand-depth/json-schema-2020-12-default-model-expand-depth.yaml [hunk 1]
```diff
@@ -15,6 +15,11 @@ paths:
           content:
             application/json:
               schema:
+                $vocabulary:
```
Comment (glowcloud) [inline test/e2e-cypress/static/pages/json-schema-2020-12-default-model-expand-depth/json-schema-2020-12-default-model-expand-depth.yaml:18]: I was not sure about the usage of `$vocabulary`, let me know if this one is fine here.

Comment (char0n) [reply test/e2e-cypress/static/pages/json-schema-2020-12-default-model-expand-depth/json-schema-2020-12-default-model-expand-depth.yaml:18]: In JSON Schema 2020-12, $vocabulary keyword has following shape:

```
  "$vocabulary": {
    "https://json-schema.org/draft/2020-12/vocab/core": true,
    "https://json-schema.org/draft/2020-12/vocab/applicator": true,
    "https://json-schema.org/draft/2020-12/vocab/validation": true,
    "https://json-schema.org/draft/2020-12/vocab/meta-data": true,
    "https://json-schema.org/draft/2020-12/vocab/format-annotation": false,
    "https://example.com/my-custom-vocab": true
  },
```

The above is just an example, but you can use  it in your fixture.

File: test/e2e-cypress/e2e/features/json-schema-2020-12-default-model-expand-depth.cy.js
Comment (char0n) [inline test/e2e-cypress/e2e/features/json-schema-2020-12-default-model-expand-depth.cy.js:1]: Let's move this to `e2e/features/plugins/json-schema-2020-12/expansion.cy.js

File: test/e2e-cypress/e2e/features/json-schema-2020-12-default-model-expand-depth.cy.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/**
+ * @prettier
+ */
+
+describe("defaultModelExpandDepth for JSON Schema 2020-12 keywords", () => {
```
Comment (char0n) [inline test/e2e-cypress/e2e/features/json-schema-2020-12-default-model-expand-depth.cy.js:5]:
```suggestion
describe("JSON Schema 2020-12 complex keywords expansion", () => {
```

File: test/e2e-cypress/e2e/features/json-schema-2020-12-default-model-expand-depth.cy.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  it("should expand schemas", () => {
```
Comment (char0n) [inline test/e2e-cypress/e2e/features/json-schema-2020-12-default-model-expand-depth.cy.js:6]:
```suggestion
  it("should deeply expand all Schemas and complex keywords", () => {
```

**Review by char0n — CHANGES_REQUESTED** [review]
Looks good! Please have a look at my CRCs

**Review by char0n — APPROVED** [review]
LGTM. Nice job!

**Pull request merged**